src/Amadevus.RecordGenerator.Generators/GeneratedCodeAttributeGenerator.cs

```diff
@@ -12,32 +12,96 @@ namespace Amadevus.RecordGenerator.Generators
 {
     public static class GeneratedCodeAttributeGeneratorExtensions
     {
-        public static TypeDeclarationSyntax WithGeneratedCodeAttribute(
-            this TypeDeclarationSyntax typeSyntax)
+        private static readonly GeneratedCodeAttributeApplier attributeApplier
+            = new GeneratedCodeAttributeApplier(new GeneratedCodeAttributeGenerator());
+
+        public static ClassDeclarationSyntax WithGeneratedCodeAttributeOnMembers(
+            this ClassDeclarationSyntax typeSyntax)
         {
-            return new GeneratedCodeAttributeGenerator(typeSyntax)
-                .GetWithGeneratedCodeAttribute();
+            return attributeApplier.GetWithGeneratedCodeAttributeOnMembers(typeSyntax);
         }
     }
 
-    internal sealed class GeneratedCodeAttributeGenerator
+    internal sealed class GeneratedCodeAttributeApplier
     {
-        private readonly TypeDeclarationSyntax typeSyntax;
+        private readonly GeneratedCodeAttributeGenerator attributeGenerator;
+
+        private readonly IDictionary<Type, Func<MemberReplacementContext, SyntaxNode>> Updates
+            = new Dictionary<Type, Func<MemberReplacementContext, SyntaxNode>>
+        {
+            {
+                typeof(MethodDeclarationSyntax),
+                (context) => ((MethodDeclarationSyntax)context.Member).WithAttributeLists(context.AttributeGenerator.GenerateAttributeListSyntaxList())
+            },
+            {
+                typeof(PropertyDeclarationSyntax),
+                (context) => ((PropertyDeclarationSyntax)context.Member).WithAttributeLists(context.AttributeGenerator.GenerateAttributeListSyntaxList())
+            },
+            {
+                typeof(ConstructorDeclarationSyntax),
+                (context) => ((ConstructorDeclarationSyntax)context.Member).WithAttributeLists(context.AttributeGenerator.GenerateAttributeListSyntaxList())
+            },
+            {
+                typeof(ClassDeclarationSyntax),
+                (context) => context.AttributeApplier.GetWithGeneratedCodeAttributeOnMembers((ClassDeclarationSyntax)context.Member)
+            }
+        };
+
+
+        public GeneratedCodeAttributeApplier(
+            GeneratedCodeAttributeGenerator attributeGenerator)
+        {
+            this.attributeGenerator = attributeGenerator;
+        }
 
-        public GeneratedCodeAttributeGenerator(
-            TypeDeclarationSyntax typeSyntax)
+        
+        public ClassDeclarationSyntax GetWithGeneratedCodeAttributeOnMembers(
+            ClassDeclarationSyntax typeDeclaration)
         {
-            this.typeSyntax = typeSyntax;
+            Func<SyntaxNode, bool> toBeReplaced = (node) =>
```

Comment:
I feel like those two lambdas would look better as local functions, but it's really a personal preference.

Reply:
Im used to the lambda style because of the code im usually working in, but don't really care.
I was able to remove one and adjust the other to a local function.